### ISSUE TITLE
[$500] Android - IOU - Validation message not shown when file above 25mb for Smart Recipe was selected

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#27389]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/27089
## Action Performed:
1. Open app
2. Tap FAB
3. Tap on Request Money
4. Select Smart Recipe tab
5. Tap on add file from gallery button
6. Select any photo larger than 25mb
7. Select any attendee
8. Tap on Request button

## Expected Result:
The validation message should be show when user tries use the photo larger than 25mb for Smart Recipe

## Actual Result:
Validation message not shown and user able to request money with smart recipe option when file is larger than 25mb

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.69-0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/38135323-af3f-4b5b-915d-6139420d8c0c

https://github.com/Expensify/App/assets/78819774/567a7272-75eb-4110-952b-954d79a64c35

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01898f68a06cf0cea0</li>
        <li>Upwork Job ID: 1703675063206494208</li>
        <li>Last Price Increase: 2023-09-18</li>
<li>Automatic offers: </li>
<ul>
<li>ishpaul777 | Contributor | 26791640</li>
</ul></li>
    </ul>
</details>

**Comment by ishpaul777** [comment]
## Proposal

### Problem 
Validation message not shown when file above 25mb for Smart Recipe was selected

## Root cause 
There is validation of file size or any other validation on after image is picked 

https://github.com/ishpaul777/App/blob/1bdc5acc12f982b4f6e37bfadf9060ac3d2d3b3c/src/pages/iou/ReceiptSelector/index.native.js#L262

### Changes 
Validate the asset object when picked by user 
```jsx
                    showImagePicker(launchImageLibrary)
                            .then((receiptImage) => {
                                if (receiptImage.fileSize > CONST.IOU.MAX_RECEIPT_FILE_SIZE) {
                                    // show error message if the file size is too large
                                }  else if (receiptImage.fileSize < CONST.IOU.MIN_RECEIPT_FILE_SIZE) {
                                    // show error message if the file size is too small
                                }
```

**Comment by sakluger** [comment]
Looks like this may be a regression from https://github.com/Expensify/App/issues/25865, I've commented in that issue for clarification.

**Comment by sakluger** [comment]
Okay it's not a regression because we didn't ever have the size validation on mobile. As part of this fix, let's make sure we add the validation across all platforms where it hasnt been added yet.

**Comment by deetergp** [comment]
Good enough for me, let's go with @ishpaul777's proposal.

**Comment by deetergp** [comment]
@ishpaul777 Any updates? Do we need to raise the price and make this generally available agin?

**Comment by deetergp** [comment]
My apologies but the thread on this one is very hard to follow. Can someone please summarize what is going on and where we are at with this issue? @ishpaul777 @allroundexperts

**Comment by situchan** [comment]
I suggest to put this on hold for https://github.com/Expensify/App/pull/29963

**Comment by situchan** [comment]
> @situchan Is #29963 solving this issue too?

I think so

**Comment by situchan** [comment]
> I meant for similar case like validation error on web has a custom modal I think?

yes because web native alert is ugly

**Comment by situchan** [comment]
If we should use custom alert for all platforms for consistency, we should fix it for all cases, not only this one.
Btw, it's new issue and should get design approval. Out of scope for this one.
